test/Table_Tests/src/IO/Delimited_Read_Spec.enso

```diff
@@ -225,13 +225,45 @@ spec =
 
         Test.specify "should fail in presence of a mismatched quote" <|
             [Problem_Behavior.Report_Error, Problem_Behavior.Report_Warning, Problem_Behavior.Ignore].each pb->
-                r1 = Data.read (enso_project.data / "mismatched_quote.csv") (Delimited "," headers=True value_formatter=Nothing) on_problems=pb
+                format = (Delimited "," headers=True value_formatter=Nothing)
+                r1 = Data.read (enso_project.data / "mismatched_quote.csv") format on_problems=pb
                 r1.should_fail_with File_Error.Corrupted_Format
                 r1.catch.to_display_text.should_contain "quote has been opened but never closed"
+                r1.catch.to_display_text.should_contain '["\n7,8,9\n]'
 
-                r2 = Data.read (enso_project.data / "mismatched_quote2.csv") (Delimited "," headers=True value_formatter=Nothing) on_problems=pb
+                r2 = Data.read (enso_project.data / "mismatched_quote2.csv") format on_problems=pb
                 r2.should_fail_with File_Error.Corrupted_Format
 
+                format3 = format . with_quotes quote_escape="\"
+                r3 = Data.read (enso_project.data / "mismatched_quote3.csv") format3 on_problems=pb
+                r3.should_fail_with File_Error.Corrupted_Format
+
+                r4 = Data.read (enso_project.data / "mismatched_quote4.csv") format on_problems=pb
+                r4.should_fail_with File_Error.Corrupted_Format
+
+                r5 = Data.read (enso_project.data / "mismatched_quote5.csv") format on_problems=pb
+                r5.should_fail_with File_Error.Corrupted_Format
+
+        Test.specify "should handle quotes if they are opened in the middle of an unquoted cell in a sane way" <|
+            t1 = Data.read (enso_project.data / "mismatched_quote_at_end.csv") (Delimited "," headers=True value_formatter=Nothing)
+            t1.column_names . should_equal ["a", "b", "c"]
+            t1.at 'a' . to_vector . should_equal ['1', 'abc', '7']
+            t1.at 'b' . to_vector . should_equal ['2', 'def', '8']
+            t1.at 'c' . to_vector . should_equal ['3', 'g h i"', '9']
+
+            t2 = Data.read (enso_project.data / "weird_quoting_stuff.csv") (Delimited "," headers=True value_formatter=Nothing)
+            t2.column_names . should_equal ["A", "B", "C", "D", "E"]
+            t2.at 'A' . to_vector . should_equal ['Te,s"t', 'Te,s"t']
+            t2.at 'B' . to_vector . should_equal ['An " other', 'An " other"']
+            t2.at 'C' . to_vector . should_equal ['He\nllo', 'He\nllo']
+            t2.at 'D' . to_vector . should_equal ['This is an escaped quote \\"', 'This is an escaped quote \\"']
+            t2.at 'E' . to_vector . should_equal ['Excel escapes "" with 8 quotes """"""', 'Excel escapes "" with 8 quotes """"""']
+
+            t3 = Data.read (enso_project.data / "weird_quoting_stuff2.csv") (Delimited "," headers=True value_formatter=Nothing)
+            IO.println t3
+            t3.print
```

Comment:
I added this test with the cases that we discussed we are not sure how they should be handled.
i.e. quotes started in the beginning of the string but closed, but then continued with further content.

```
A,B,C
1,2,3
"Hello!" shouted Bob,10,20
4,5,6
Jack "the Ripper" was here,30,40
```

This currently fails with:

```
An unexpected error was returned: (Corrupted_Format (File C:\NBO\enso\test\Table_Tests\data\weird_quoting_stuff2.csv) 'A quote has been opened but never closed inside of a Delimited file, so it cannot be interpreted properly. Inside cell: ["Hello!" shouted Bob].' (Mismatched_Quote.Error '"Hello!" shouted Bob'))
```

Reply:
So, I'm a bit on the fence here.

Aside from this little test, all the others work (tbh I surprised because my instinct tells me the logic should not work correctly for `mismatched_quotes3.csv`...).

cc: @jdunkerley I know you told me that we are not testing for this one because we do not know what the right behaviour should be. I still wanted to see what will happen. I'm wondering if we should try to improve this slightly, because currently this is a little bit weird - the value doesn't really have a mismatched quote but the _Mismatched Quote_ error is returned. I'm wondering if we should try to recover, ignore this or try returning a different kind of error here? What do you think? Let's discuss this on Monday.